frontends/web/src/components/forms/select.tsx

```diff
@@ -14,26 +15,40 @@
  * limitations under the License.
  */
 
-import { h } from 'preact';
+import { h, JSX } from 'preact';
 import { select } from './select.css';
 
-export default function Select({
-    id,
-    label = null,
+type TOptionTextContent = {
+    text: string;
+}
+
+type TOption = JSX.IntrinsicElements['option'] & TOptionTextContent
+
+type TSelectProps = {
+    // Temp add defaultValue, see https://github.com/preactjs/preact/issues/2668
+    defaultValue?: string;
+    id: string;
+    label?: string;
+    options: TOption[];
+    selectedOption?: string;
+} & JSX.IntrinsicElements['select']
+
+export function Select({
+    label,
     options = [],
-    selected = null,
+    selectedOption,
     ...props
-}) {
+}: TSelectProps) {
     return (
         <div className={select}>
-            {label && <label for={id}>{label}</label>}
-            <select id={id} {...props}>
+            {label && <label for={props.id}>{label}</label>}
```

Comment:
why do `...props` and then `props.id` instead of putting `id` next to the other props above?

Reply:
could yes, but then it also needs to be added to the next line, i.e `<select {...props}>` would change to `<select id={id} {...props}>`

Reply:
up to you!

Reply:
fixed in https://github.com/digitalbitbox/bitbox-wallet-app/compare/3f5ec258e7b3a9aa0cd38e325bf4abe03b2a9f38..360ce5c0aec87ee2b03c40428be86d79a39413a9 also made separate commit and moved select.jsx to tsx first to preserve the history.